src/nvim/api/vim.c

```diff
@@ -2116,11 +2120,21 @@ Dictionary nvim_eval_statusline(String str, Dict(eval_statusline) *opts, Error *
       return result;
     }
   }
-  VALIDATE(!(use_winbar && use_tabline), "%s", "Cannot use both 'use_winbar' and 'use_tabline'", {
+  if (HAS_KEY(opts->use_statuscol)) {
+    use_statuscol = api_object_to_bool(opts->use_statuscol, "use_statuscol", false, err);
+
+    if (ERROR_SET(err)) {
+      return result;
+    }
+  }
+  VALIDATE(!(use_winbar && use_tabline && use_statuscol), "%s",
+           "Can only use one of 'use_winbar', 'use_tabline' and 'use_statuscol'", {
     return result;
   });
```

Comment:
This doesn't give an error if two of them are used.

Reply:
Right, whoops.